homeassistant/components/sensor/fronius.py

```diff
@@ -0,0 +1,158 @@
+"""
+Support for Fronius devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.fronius/
+"""
+import asyncio
+from concurrent.futures._base import TimeoutError
+from datetime import timedelta
+import logging
+
+from aiohttp.client_exceptions import ServerDisconnectedError
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+import voluptuous as vol
+
+REQUIREMENTS = ['pyfronius==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TYPE = 'type'
+CONF_DEVICEID = 'device'
+CONF_SCOPE = 'scope'
+
+TYPE_INVERTER = 'inverter'
+TYPE_STORAGE = 'storage'
+TYPE_METER = 'meter'
+TYPE_POWER_FLOW = 'power_flow'
+SCOPE_DEVICE = 'device'
+SCOPE_SYSTEM = 'system'
+
+DEFAULT_SCOPE = SCOPE_DEVICE
+DEFAULT_DEVICE = None
+
+SENSOR_TYPES = [TYPE_INVERTER, TYPE_STORAGE, TYPE_METER, TYPE_POWER_FLOW]
+SCOPE_TYPES = [SCOPE_DEVICE, SCOPE_SYSTEM]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TYPE): vol.In(SENSOR_TYPES),
+    vol.Optional(CONF_SCOPE, default=DEFAULT_SCOPE):
+        vol.All(cv.ensure_list, [vol.In(SCOPE_TYPES)]),
+    vol.Optional(CONF_DEVICEID): cv.positive_int,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up of Fronius platform."""
+    _LOGGER.debug(config)
+    import pyfronius
+
+    session = async_get_clientsession(hass)
+    fronius = pyfronius.Fronius(session, config[CONF_HOST])
+
+    name = "fronius_{}_{}".format(config[CONF_TYPE], config[CONF_HOST])
+    if CONF_DEVICEID in config.keys():
```

Comment:
No need for `keys()`

Reply:
I'll try it, but AFAIR without keys() something different happened.